python/lsst/afw/table/catalogMatches.py

```diff
@@ -0,0 +1,164 @@
+# 
+# LSST Data Management System
+# Copyright 2008-2015 AURA/LSST.
+# 
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the LSST License Statement and 
+# the GNU General Public License along with this program.  If not, 
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+from __future__ import absolute_import
+
+import os.path
+
+from .tableLib import BaseCatalog, SimpleCatalog, SourceCatalog, SimpleTable, SourceTable, \
+                      Schema, ReferenceMatch
+from lsst.utils import getPackageDir
+
+__all__ = ["copySchema", "copyCatalog", "matchesToCatalog", "matchesFromCatalog"]
+
+def copySchema(schema, target, targetPrefix=None, sourcePrefix=None):
+    """Return a deep copy the provided schema
```

Comment:
"... deep copy *of* the ..."

Also, could you please clarify in the docstring what the relationship between `target` and the return value is? Can the caller ignore the return value and assume that whatever they passed as `target` is modified by this function?

Also also: would it be possible to make this function and `copyCatalog`, below, considerably simpler & less verbose by using a `SchemaMapper`? I spent a few minutes playing around and came up with https://gist.github.com/jdswinbank/a2418d7c8a00c9c1f6ea, which provides replacements for these functions which are quite a lot shorter. They pass the `SourceMatchTestCase`, below, but haven't otherwise been sanity checked, so I might've missed some subtlety and could probably simplify further with a bit more effort.